### ISSUE TITLE
OneBusAway Android app drains my battery after being opened, and doesn't stop until I force stop the app in Android's app settings.

### ISSUE THREAD

**Issue by smkent** [OneBusAway/onebusaway-android#1016]
**Summary:** 

OneBusAway on Android drains the battery after starting, even when no longer in use. I have to force stop the OneBusAway app in Android's app settings to stop the battery drain.

I'm an occasional transit user and I have noticed this every time I've used OneBusAway over the past few months. I always have to remember to force stop OneBusAway so my battery doesn't drain. I took the bus this morning, forgot to stop the app after getting on, and once again my phone is almost dead with Android's battery settings showing OneBusAway as the cause (screenshot at the end of this issue report).

I'm not sure if OBA's configured data source matters, but I am in Puget Sound.

**Steps to reproduce:** 

1) Open OneBusAway.
2) Switch to another app or return to the home screen.
3) Leave your phone unplugged for a while (an hour or two should be sufficient)
4) Observe significant battery drain, with Android's battery app showing OneBusAway as the culprit

**Expected behavior:** 

OneBusAway shouldn't rapidly drain my battery

**Observed behavior:** 

OneBusAway rapidly drains my battery

**Device and Android version:** 

I have a Moto G4 Plus, with OneBusAway 2.5.5.

**Screenshot:** 

![onebusaway-battery](https://user-images.githubusercontent.com/2831985/66354678-7d2c7d80-e91a-11e9-92f8-ca6983e1d9b0.png)

**Comment by smkent** [comment]
Tapping into the battery details page, it appears GPS was on the entire time the OneBusAway app was running until I force stopped it:

![battery-details](https://user-images.githubusercontent.com/2831985/66356067-8586b780-e91e-11e9-80b7-76d75f60a35f.png)

This is definitely a bug as GPS shouldn't be needed unless the user is using the map within the app.

**Comment by barbeau** [comment]
@smkent What version of Android are you running, and what version of Google Play Services?  If you do "Send feedback->App feedback" you can copy/paste your device info from there. Also, are you running an original factory ROM on your device or a custom ROM?

This is a duplicate of https://github.com/OneBusAway/onebusaway-android/issues/582.  I'd like to copy your device info into that issue and close this one to keep all the information in the same place.

In short, this seems to be a bug in Google Play Services that occasionally pops up. OneBusAway uses the fused location provider which is implemented by Google in Google Play Services. Unfortunately there isn't much we can do about it, other than collect data to share with Google in a request to fix the problem.

**Comment by smkent** [comment]
> What version of Android are you running, and what version of Google Play Services? If you do "Send feedback->App feedback" you can copy/paste your device info from there. Also, are you running an original factory ROM on your device or a custom ROM?

I am running the original factory ROM.

Here is my device info:

> App Version: 2.5.5
> Model: Moto G (4)
> OS Version: 7.0 / 24
> Google Play Services App: 19.2.75 (040308-269183835)
> Google Play Services Library: 12451000
> Region/API: Puget Sound (selected automatically)

**Comment by barbeau** [comment]
@smkent Thanks!  I've copied this info over to https://github.com/OneBusAway/onebusaway-android/issues/582, so feel free to follow progress there.
